### ISSUE TITLE
Failed to restore Gradle cache entry > 2Gb due to ERR_OUT_OF_RANGE

### ISSUE THREAD

**Issue by safa007** [gradle/gradle-build-action#205]
I get this warning when attempting to restore gradle cache after the cache has been downloaded.

```
Warning: Failed to restore v6-gradle|macOS|ui-test[37a6259cc0c1dae299a7866489dff0bd]-984a5b4d4759dc43ffeac3a41c3bcdf113bd34fc: RangeError [ERR_OUT_OF_RANGE]: The value of "length" is out of range. It must be >= 0 && <= 2147483647. Received 2216518188
```

It looks like integer is not big enough to represent the size of the gradle cache. Could we change it to use a long instead?

**Comment by bigdaz** [comment]
The `gradle-build-action` doesn't define any variable named `length` so it's not a simple matter of changing a declared type.  There are a few things you could do to help track down where this is coming from:
1. Run the build with [Action Step debugging enabled](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging#enabling-step-debug-logging) and share the log output.
2. Even better, share a link to the actual GitHub Actions run.
3. Try out the just-released v2.1.4, which will log the stack trace at debug level, or at info level if you [enable debugging for the `gradle-build-action`](https://github.com/gradle/gradle-build-action#cache-debugging-and-analysis)

Note that there are 2 different ways you can enable debugging. The first "Action Step debugging" requires you to set a secret on the repository and will show debug logging from this action and all other sources. The second "Cache debugging" involves setting an environment variable and results in this action writing debug messages at info level.

**Comment by safa007** [comment]
Sure, here's the log output:

```
##[debug]Evaluating condition for step: 'Gradle cache'
##[debug]Evaluating: success()
##[debug]Evaluating success:
##[debug]=> true
##[debug]Result: true
##[debug]Starting: Gradle cache
##[debug]Register post job cleanup for action: gradle/gradle-build-action@v[2](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:2)
##[debug]Loading inputs
##[debug]Evaluating: toJSON(matrix)
##[debug]Evaluating toJSON:
##[debug]..Evaluating matrix:
##[debug]..=> null
##[debug]=> 'null'
##[debug]Result: 'null'
##[debug]Loading env
Run gradle/gradle-build-action@v2
  with:
    cache-disabled: false
    cache-read-only: false
    gradle-home-cache-includes: caches
  notifications
  
    cache-write-only: false
    gradle-home-cache-strict-match: false
    workflow-job-context: null
  env:
    JAVA_HOME_11.0.14_x64: /Users/runner/hostedtoolcache/jdk/11.0.14/x64
    JAVA_HOME: /Users/runner/hostedtoolcache/jdk/11.0.14/x64
    JAVA_HOME_11_0_14_X64: /Users/runner/hostedtoolcache/jdk/11.0.14/x64
::save-state name=GRADLE_BUILD_ACTION_CACHE_RESTORED::true
##[debug]Save intra-action state GRADLE_BUILD_ACTION_CACHE_RESTORED = true
::save-state name=GRADLE_USER_HOME::/Users/runner/.gradle
##[debug]Save intra-action state GRADLE_USER_HOME = /Users/runner/.gradle
::group::Restore Gradle state from cache
Restore Gradle state from cache
  ##[debug]Requesting Gradle User Home with
  ##[debug]    key:v6-gradle|macOS|ui-test[[3](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:3)7a6259cc0c1dae299a7866[4](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:4)89dff0bd]-984a[5](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:5)b4d4759dc43ffeac3a41c3bcdf113bd34fc
  ##[debug]    restoreKeys:[v[6](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:6)-gradle|macOS|ui-test[3[7](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:7)a6259cc0c1dae299a7[8](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:8)6648[9](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:9)dff0bd],v6-gradle|macOS|ui-test,v6-gradle|macOS]
  ##[debug]Using cache paths: /Users/runner/.gradle/caches,/Users/runner/.gradle/notifications,/Users/runner/.gradle/.gradle-build-action
  ##[debug]Resolved Keys:
  ##[debug]["v6-gradle|macOS|ui-test[37a6259cc0c1dae299a7866489dff0bd]-984a5b4d4759dc43ffeac3a41c3bcdf113bd34fc","v6-gradle|macOS|ui-test[37a6259cc0c1dae299a7866489dff0bd]","v6-gradle|macOS|ui-test","v6-gradle|macOS"]
  ##[debug]Checking zstd --version
  ##[debug]*** zstd command line interface 64-bits v1.5.2, by Yann Collet ***
  ##[debug]Resource Url: https://artifactcache.actions.githubusercontent.com/d6NZdvDA0IohzglwQViWZ6MN171Wm8sWoGBC9TIuabnD0j77G2/_apis/artifactcache/cache?keys=v6-gradle%7CmacOS%7Cui-test%5B37a6259cc0c1dae299a7866489dff0bd%5D-984a5b4d4759dc43ffeac3a41c3bcdf113bd34fc%2Cv6-gradle%7CmacOS%7Cui-test%5B37a6259cc0c1dae299a7866489dff0bd%5D%2Cv6-gradle%7CmacOS%7Cui-test%2Cv6-gradle%7CmacOS&version=ebce7290c324ba9d5e6ef4959d2adb3ff4b86f302096a[10](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:10)0ef31e0cbd9f10632
  ::add-mask::***
  ##[debug]Cache Result:
  ##[debug]{"scope":"refs/pull/6958/merge","cacheKey":"v6-gradle|macOS|ui-test[37a6259cc0c1dae299a7866489dff0bd]-795bf3de888e338f92eff8[11](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:11)bfb6db93c02f3ed8","cacheVersion":"ebce7290c324ba9d5e6ef4959d2adb3ff4b86f302096a100ef31e0cbd9f10632","creationTime":"2022-03-18T01:09:31.1666667Z","archiveLocation":"***"}
  ##[debug]Archive Path: /Users/runner/work/_temp/7a0af9f5-9136-44ef-862a-c71761d99956/cache.tzst
  ##[debug]Use Azure SDK: true
  ##[debug]Download concurrency: 8
  ##[debug]Request timeout (ms): 30000
  ##[debug]Downloading segment at offset 0 with length 2216518188...
  Received 0 of 2216518188 (0.0%), 0.0 MBs/sec
  Received 37748736 of 2216518188 (1.7%), 18.0 MBs/sec
  Received 83886080 of 2216518188 (3.8%), 26.6 MBs/sec
  Received 134217728 of 2216518188 (6.1%), 31.9 MBs/sec
  Received 192937984 of 2216518188 (8.7%), 36.7 MBs/sec
  Received 251658240 of 2216518188 (11.4%), 39.6 MBs/sec
  Received 301989888 of 2216518188 (13.6%), 40.8 MBs/sec
  Received 343932928 of 2216518188 (15.5%), 40.7 MBs/sec
  Received 390070272 of 2216518188 (17.6%), 41.0 MBs/sec
  Received 411041792 of 2216518188 (18.5%), 38.9 MBs/sec
  Received 452984832 of 2216518188 (20.4%), 39.0 MBs/sec
  Received 486539264 of 2216518188 (22.0%), 38.4 MBs/sec
  Received 520093696 of 2216518188 (23.5%), 37.9 MBs/sec
  Received 557842432 of 2216518188 (25.2%), 37.8 MBs/sec
  Received 599785472 of 2216518188 (27.1%), 37.9 MBs/sec
  Received 6417285[12](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:12) of 2216518188 (29.0%), 38.1 MBs/sec
  Received 675282944 of 2216518188 (30.5%), 37.7 MBs/sec
  Received 7[13](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:13)031680 of 2216518188 (32.2%), 37.6 MBs/sec
  Received 763363328 of 2216518188 (34.4%), 38.1 MBs/sec
  Received 801112064 of 2216518188 (36.1%), 38.0 MBs/sec
  Received 847249408 of 2216518188 (38.2%), 38.3 MBs/sec
  Received 889192448 of 2216518188 (40.1%), 38.4 MBs/sec
  Received 939524096 of 2216518188 (42.4%), 38.8 MBs/sec
  Received 977272832 of 2216518188 (44.1%), 38.7 MBs/sec
  Received 1023410176 of 2216518188 (46.2%), 38.9 MBs/sec
  Received 1069547520 of 2216518188 (48.3%), 39.1 MBs/sec
  Received 1119879168 of 2216518188 (50.5%), 39.4 MBs/sec
  Received 1153433600 of 2216518188 (52.0%), 39.1 MBs/sec
  Received 1191182336 of 2216518188 (53.7%), 39.0 MBs/sec
  Received 1241513984 of 2216518188 (56.0%), 39.3 MBs/sec
  Received 1291845632 of 2216518188 (58.3%), 39.6 MBs/sec
  Received 1321205760 of 2216518188 (59.6%), 39.2 MBs/sec
  Received 1367343104 of 2216518188 (61.7%), 39.4 MBs/sec
  Received 13883[14](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:14)624 of 2216518188 (62.6%), 38.8 MBs/sec
  Received 1426063360 of 2216518188 (64.3%), 38.7 MBs/sec
  Received 1463812096 of 2216518188 (66.0%), 38.6 MBs/sec
  Received [15](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:15)01560832 of 22[16](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:16)518188 (67.7%), 38.6 MBs/sec
  Received 1543503872 of 2216518188 (69.6%), 38.6 MBs/sec
  Received 1585446912 of 2216518188 (71.5%), 38.6 MBs/sec
  Received 1614807040 of 2216518188 (72.9%), 38.4 MBs/sec
  Received 1652555776 of 2216518188 (74.6%), 38.3 MBs/sec
  Received 1694498816 of 2216518188 (76.4%), 38.4 MBs/sec
  Received [17](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:17)28053248 of 22165[18](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:18)188 (78.0%), 38.2 MBs/sec
  Received 176580[19](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:19)84 of 2216518188 (79.7%), 38.2 MBs/sec
  Received 1807745024 of 2216518188 (81.6%), 38.2 MBs/sec
  Received 1837105152 of 2216518188 (82.9%), 38.0 MBs/sec
  Received 1883242496 of 2216518188 (85.0%), 38.1 MBs/sec
  Received 19[20](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:20)991232 of 2[21](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:21)6518188 (86.7%), 38.1 MBs/sec
  Received 1958739968 of [22](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:22)16518188 (88.4%), 38.0 MBs/sec
  Received 1988100096 of 2216518188 (89.7%), 37.8 MBs/sec
  Received 2017460224 of 2216518188 (91.0%), 37.6 MBs/sec
  Received 2059403264 of 2216518188 (92.9%), 37.7 MBs/sec
  Received 2088763392 of 2216518188 (94.2%), 37.5 MBs/sec
  Received 2126512128 of 2216518188 (95.9%), 37.4 MBs/sec
  Received 2164260864 of 2216518188 (97.6%), 37.4 MBs/sec
  Received 2197815296 of 2216518188 (99.2%), 37.3 MBs/sec
  Received 221[23](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:23)23884 of 2216518188 (99.8%), 36.9 MBs/sec
  Received 2216518188 of 2216518188 (100.0%), 36.8 MBs/sec
  Warning: Failed to restore v6-gradle|macOS|ui-test[37a6[25](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:25)9cc0c1dae[29](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:29)9a7866489dff0bd]-984a5b4d4759dc43ffeac3a41c3bcdf113bd[34](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:34)fc: RangeError [ERR_OUT_OF_RANGE]: The value of "length" is out of range. It must be >= 0 && <= 214748[36](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:36)47. Received 2216518188
  Gradle User Home cache not found. Will initialize empty.
  ::save-state name=CACHE_LISTENER::{"cacheEntries":[{"entryName":"Gradle User Home","requestedKey":"v6-gradle|macOS|ui-test[[37](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:37)a6259cc0c1dae299a7866489dff0bd]-984a5b4d4759dc43ffeac3a[41](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:41)c3bcdf113bd34fc","requestedRestoreKeys":["v6-gradle|macOS|ui-test[37a6259cc0c1dae299a7866489dff0bd]","v6-gradle|macOS|ui-test","v6-gradle|macOS"]}]}
  ##[debug]Save intra-action state CACHE_LISTENER = {"cacheEntries":[{"entryName":"Gradle User Home","requestedKey":"v6-gradle|macOS|ui-test[37a6259cc0c1dae299a7866489dff0bd]-984a5b4d4759dc[43](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:43)ffeac3a41c3bcdf113bd34fc","requestedRestoreKeys":["v6-gradle|macOS|ui-test[37a6259cc0c1dae299a7866[48](https://github.com/turo/marvin/runs/5602785050?check_suite_focus=true#step:4:48)9dff0bd]","v6-gradle|macOS|ui-test","v6-gradle|macOS"]}]}
  ::endgroup::
##[debug]Node Action run completed with exit code 0
##[debug]GRADLE_BUILD_ACTION_CACHE_RESTORED='true'
##[debug]Finishing: Gradle cache
```

**Comment by bigdaz** [comment]
Thank you for the log @safa007 . As you suspected, it looks like the file size is being read into a variable that doesn't have the capacity. I suspect this is happening in the `@actions/cache` library provided by GitHub Actions, but it's hard to be certain.

Was the above log produced with [v2.1.4](https://github.com/gradle/gradle-build-action/releases/tag/v2.1.4)? I added some additional logging to print the stack trace of the error, so it's not clear if the stack trace is empty or if you're running with an earlier version. (I only released v2.1.4 earlier today).

**Comment by safa007** [comment]
@bigdaz here's the updated stack trace with the additional logging

```
##[debug]Evaluating condition for step: 'Gradle cache'
##[debug]Evaluating: success()
##[debug]Evaluating success:
##[debug]=> true
##[debug]Result: true
##[debug]Starting: Gradle cache
##[debug]Register post job cleanup for action: gradle/gradle-build-action@v[2](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:2)
##[debug]Loading inputs
##[debug]Evaluating: toJSON(matrix)
##[debug]Evaluating toJSON:
##[debug]..Evaluating matrix:
##[debug]..=> null
##[debug]=> 'null'
##[debug]Result: 'null'
##[debug]Loading env
Run gradle/gradle-build-action@v2
::save-state name=GRADLE_BUILD_ACTION_CACHE_RESTORED::true
##[debug]Save intra-action state GRADLE_BUILD_ACTION_CACHE_RESTORED = true
::save-state name=GRADLE_USER_HOME::/Users/runner/.gradle
##[debug]Save intra-action state GRADLE_USER_HOME = /Users/runner/.gradle
::group::Restore Gradle state from cache
Restore Gradle state from cache
  ##[debug]Requesting Gradle User Home with
  ##[debug]    key:v6-gradle|macOS|ui-test[[3](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:3)7a6259cc0c1dae299a7866[4](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:4)89dff0bd]-6b3b418bda3000aae80b2[5](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:5)4d8b79de9340d728[6](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:6)3
  ##[debug]    restoreKeys:[v6-gradle|macOS|ui-test[3[7](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:7)a6259cc0c1dae299a7[8](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:8)6648[9](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:9)dff0bd],v6-gradle|macOS|ui-test,v6-gradle|macOS]
  ##[debug]Using cache paths: /Users/runner/.gradle/caches,/Users/runner/.gradle/notifications,/Users/runner/.gradle/.gradle-build-action
  ##[debug]Resolved Keys:
  ##[debug]["v6-gradle|macOS|ui-test[37a6259cc0c1dae299a7866489dff0bd]-6b3b418bda3000aae80b254d8b79de9340d72863","v6-gradle|macOS|ui-test[37a6259cc0c1dae299a7866489dff0bd]","v6-gradle|macOS|ui-test","v6-gradle|macOS"]
  ##[debug]Checking zstd --version
  ##[debug]*** zstd command line interface 64-bits v1.5.2, by Yann Collet ***
  ##[debug]Resource Url: https://artifactcache.actions.githubusercontent.com/d6NZdvDA0IohzglwQViWZ6MN171Wm8sWoGBC9TIuabnD0j77G2/_apis/artifactcache/cache?keys=v6-gradle%7CmacOS%7Cui-test%5B37a6259cc0c1dae299a7866489dff0bd%5D-6b3b418bda3000aae80b254d8b79de9340d72863%2Cv6-gradle%7CmacOS%7Cui-test%5B37a6259cc0c1dae299a7866489dff0bd%5D%2Cv6-gradle%7CmacOS%7Cui-test%2Cv6-gradle%7CmacOS&version=ebce7290c324ba9d5e6ef4959d2adb3ff4b86f302096a[10](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:10)0ef31e0cbd9f10632
  ::add-mask::***
  ##[debug]Cache Result:
  ##[debug]{"scope":"refs/pull/6958/merge","cacheKey":"v6-gradle|macOS|ui-test[37a6259cc0c1dae299a7866489dff0bd]-2914862981442acb89f56e1da35345fae8fad0ff","cacheVersion":"ebce7290c324ba9d5e6ef4959d2adb3ff4b86f302096a100ef31e0cbd9f10632","creationTime":"2022-03-18T21:12:59.7166667Z","archiveLocation":"***"}
  ##[debug]Archive Path: /Users/runner/work/_temp/258c3cae-6b8a-4d10-b650-d1c7caad0e41/cache.tzst
  ##[debug]Use Azure SDK: true
  ##[debug]Download concurrency: 8
  ##[debug]Request timeout (ms): 30000
  ##[debug]Downloading segment at offset 0 with length 2718006789...
  Received 0 of 2718006789 (0.0%), 0.0 MBs/sec
  Received 33554432 of 2718006789 (1.2%), 16.0 MBs/sec
  Received 92274688 of 2718006789 (3.4%), 29.2 MBs/sec
  Received 138412032 of 2718006789 (5.1%), 32.9 MBs/sec
  Received 192937984 of 2718006789 (7.1%), 36.7 MBs/sec
  Received 243269632 of 2718006789 (9.0%), 38.5 MBs/sec
  Received 301989888 of 2718006789 ([11](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:11).1%), 41.0 MBs/sec
  Received 352321536 of 2718006789 (13.0%), 41.9 MBs/sec
  Received 390070272 of 2718006789 (14.4%), 41.2 MBs/sec
  Received 427819008 of 2718006789 (15.7%), 40.7 MBs/sec
  Received 469762048 of 2718006789 (17.3%), 40.6 MBs/sec
  Received 499[12](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:12)2176 of 2718006789 (18.4%), 39.6 MBs/sec
  Received 536870912 of 2718006789 (19.8%), 39.3 MBs/sec
  Received 562036736 of 2718006789 (20.7%), 38.2 MBs/sec
  Received 595591168 of 2718006789 (21.9%), 37.7 MBs/sec
  Received 629145600 of 2718006789 (23.1%), 37.4 MBs/sec
  Received 658505728 of 2718006789 (24.2%), 36.8 MBs/sec
  Received 679477248 of 2718006789 (25.0%), 35.9 MBs/sec
  Received 700448768 of 2718006789 (25.8%), 35.1 MBs/sec
  Received 734003200 of 2718006789 (27.0%), 34.9 MBs/sec
  Received 754974720 of 2718006789 (27.8%), 34.2 MBs/sec
  Received 792723456 of 2718006789 (29.2%), 34.3 MBs/sec
  Received 822083584 of 2718006789 (30.2%), 34.0 MBs/sec
  Received 843055104 of 2718006789 (31.0%), 33.4 MBs/sec
  Received 868220928 of 2718006789 (31.9%), 33.0 MBs/sec
  Received 901775360 of 2718006789 (33.2%), 33.0 MBs/sec
  Received 926941184 of 2718006789 (34.1%), 32.6 MBs/sec
  Received 960495616 of 2718006789 (35.3%), 32.6 MBs/sec
  Received 994050048 of 2718006789 (36.6%), 32.6 MBs/sec
  Received 1027604480 of 2718006789 (37.8%), 32.6 MBs/sec
  Received 1052770304 of 2718006789 (38.7%), 32.3 MBs/sec
  Received 1086324736 of 2718006789 (40.0%), 32.3 MBs/sec
  Received 1119879168 of 2718006789 (41.2%), 32.3 MBs/sec
  Received 1157627904 of 2718006789 (42.6%), 32.4 MBs/sec
  Received 1199570944 of 2718006789 (44.1%), 32.6 MBs/sec
  Received 1224736768 of 2718006789 (45.1%), 32.4 MBs/sec
  Received 1249902592 of 2718006789 (46.0%), 32.1 MBs/sec
  Received 128765[13](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:13)28 of 2718006789 (47.4%), 32.2 MBs/sec
  Received 1321205760 of 2718006789 (48.6%), 32.2 MBs/sec
  Received 1354760192 of 2718006789 (49.8%), 32.2 MBs/sec
  Received 1384120320 of 2718006789 (50.9%), 32.1 MBs/sec
  Received [14](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:14)13480448 of 2718006789 (52.0%), 32.0 MBs/sec
  Received 1451229184 of 2718006789 (53.4%), 32.1 MBs/sec
  Received 1488977920 of 2718006789 (54.8%), 32.2 MBs/sec
  Received [15](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:15)18338048 of 2718006789 (55.9%), 32.1 MBs/sec
  Received 1547698176 of 2718006789 (56.9%), 32.0 MBs/sec
  Received 1581252608 of 2718006789 (58.2%), 32.0 MBs/sec
  Received [16](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:16)23195648 of 2718006789 (59.7%), 32.1 MBs/sec
  Received 1660944384 of 2718006789 (61.1%), 32.2 MBs/sec
  Received 1702887424 of 2718006789 (62.7%), 32.4 MBs/sec
  Received 1728053248 of 2718006789 (63.6%), 32.2 MBs/sec
  Received 1769996288 of 2718006789 (65.1%), 32.4 MBs/sec
  Received 1807745024 of 2718006789 (66.5%), 32.4 MBs/sec
  Received 1837105152 of 2718006789 (67.6%), 32.3 MBs/sec
  Received 1858076672 of 2718006789 (68.4%), 32.1 MBs/sec
  Received 1904214016 of 2718006789 (70.1%), 32.3 MBs/sec
  Received 1933574144 of 2718006789 (71.1%), 32.3 MBs/sec
  Received 1962934272 of 2718006789 (72.2%), 32.2 MBs/sec
  Received 1996488704 of 2718006789 (73.5%), 32.2 MBs/sec
  Received 2034237440 of 2718006789 (74.8%), 32.2 MBs/sec
  Received 2071986176 of 2718006789 (76.2%), 32.3 MBs/sec
  Received 2109734912 of 2718006789 (77.6%), 32.4 MBs/sec
  Received 2155872256 of 2718006789 (79.3%), 32.5 MBs/sec
  Received 2193620992 of 2718006789 (80.7%), 32.6 MBs/sec
  Received 2231369728 of 2718006789 (82.1%), 32.6 MBs/sec
  Received 2269118464 of 2718006789 (83.5%), 32.7 MBs/sec
  Received 2[30](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:30)6867200 of 2718006789 (84.9%), 32.7 MBs/sec
  Received 2344615936 of 2718006789 (86.3%), 32.8 MBs/sec
  Received 2373976064 of 2718006789 (87.3%), 32.7 MBs/sec
  Received 2407530496 of 2718006789 (88.6%), 32.7 MBs/sec
  Received 2457862144 of 2718006789 (90.4%), 32.9 MBs/sec
  Received 2495610880 of 2718006789 (91.8%), 33.0 MBs/sec
  Received 2529165[31](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:31)2 of 2718006789 (93.1%), [32](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:32).9 MBs/sec
  Received 2554[33](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:33)1136 of 2718006789 (94.0%), 32.8 MBs/sec
  Received 2600468480 of 2718006789 (95.7%), 33.0 MBs/sec
  Received 26[34](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:34)022912 of 2718006789 (96.9%), 33.0 MBs/sec
  Received 2675965952 of 2718006789 (98.5%), 33.0 MBs/sec
  Received 2709618181 of 2718006789 (99.7%), 33.0 MBs/sec
  Received 2713812485 of 2718006789 (99.8%), 32.7 MBs/sec
  Received 2718006789 of 2718006789 (100.0%), 32.5 MBs/sec
  Warning: Failed to restore v6-gradle|macOS|ui-test[37a6259cc0c1dae299a7866489dff0bd]-6b3b418bda3000aae80b254d8b79de9340d72863: RangeError [ERR_OUT_OF_RANGE]: The value of "length" is out of range. It must be >= 0 && <= 214748[36](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:36)47. Received 2718006789
  ##[debug]RangeError [ERR_OUT_OF_RANGE]: The value of "length" is out of range. It must be >= 0 && <= 2147483647. Received 2718006789
  ##[debug]    at Object.writeSync (node:fs:872:5)
  ##[debug]    at Object.writeFileSync (node:fs:2159:26)
  ##[debug]    at Object.onProgress (/Users/runner/work/_actions/gradle/gradle-build-action/v2/dist/node_modules/@actions/cache/lib/internal/downloadUtils.js:217:1)
  ##[debug]    at Generator.next (<anonymous>)
  ##[debug]    at fulfilled (/Users/runner/work/_actions/gradle/gradle-build-action/v2/dist/node_modules/@actions/cache/lib/internal/downloadUtils.js:5:1)
  ##[debug]    at runMicrotasks (<anonymous>)
  ##[debug]    at processTicksAndRejections (node:internal/process/task_queues:96:5)
  Gradle User Home cache not found. Will initialize empty.
  ::save-state name=CACHE_LISTENER::{"cacheEntries":[{"entryName":"Gradle User Home","requestedKey":"v6-gradle|macOS|ui-test[[37](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:37)a6259cc0c1dae299a7866489dff0bd]-6b3b418bda3000aae80b254d8b79de93[40](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:40)d72863","requestedRestoreKeys":["v6-gradle|macOS|ui-test[37a6259cc0c1dae299a7866489dff0bd]","v6-gradle|macOS|ui-test","v6-gradle|macOS"]}]}
  ##[debug]Save intra-action state CACHE_LISTENER = {"cacheEntries":[{"entryName":"Gradle User Home","requestedKey":"v6-gradle|macOS|ui-test[37a6259cc0c1dae299a7866489dff0bd]-6b3b[41](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:41)8bda3000aae80b254d8b79de9340d72863","requestedRestoreKeys":["v6-gradle|macOS|ui-test[37a6259cc0c1dae299a7866[48](https://github.com/turo/marvin/runs/5631279645?check_suite_focus=true#step:4:48)9dff0bd]","v6-gradle|macOS|ui-test","v6-gradle|macOS"]}]}
  ::endgroup::
##[debug]Node Action run completed with exit code 0
##[debug]GRADLE_BUILD_ACTION_CACHE_RESTORED='true'
##[debug]Finishing: Gradle cache
```

**Comment by bigdaz** [comment]
Thanks. It looks like this is coming from deep in the bowels of `node.js`, which is in turn called by the `@actions/cache` library provided by GitHub. https://cs.github.com/nodejs/node/blob/a01302b8dfaa9aa6290cd6bee552c4ce30dca34c/lib/fs.js#L875

I suspect that this is due to a Gradle Home cache entry that was previously written with a size of around 2.7Gb. This is pretty huge! 
- Can you confirm this by looking at the logs for the job that uploaded the entry? You'll find it in the `Post Gradle cache` step, [as seen here](https://github.com/gradle/gradle-build-action/runs/5613811515?check_suite_focus=true#step:7:29) 
- If you have the `GRADLE_BUILD_ACTION_CACHE_DEBUG_ENABLED` variable set for your job, you should also be able to see the makeup of your Gradle User Home [as seen here](https://github.com/gradle/gradle-build-action/runs/5613809838?check_suite_focus=true#step:7:4).

If that's the case, it would appear that there's a hard limit to the size of entry that can be restored from the GitHub actions cache, and it may be possible for the `gradle-build-action` action to avoid storing these in the first place.

**Comment by safa007** [comment]
> Can you confirm this by looking at the logs for the job that uploaded the entry?
Yes it looks like a cache entry of 2.7 Gb was previously written.

I'm not exactly sure why our gradle cache is so big. Would it be worthwhile to delete that cache entry somehow and re-run the action? It seems like sometimes the cache is much smaller (700 Mb - 1 Gb) and the action succeeds in those cases.

> If you have the GRADLE_BUILD_ACTION_CACHE_DEBUG_ENABLED variable set for your job, you should also be able to see the makeup of your Gradle User Home [as seen here](https://github.com/gradle/gradle-build-action/runs/5613809838?check_suite_focus=true#step:7:4).

I've enabled that variable and can see that it is true in the debug logs, but the makeup of the gradle user home was not output.

**Comment by bigdaz** [comment]
Yes, it's definitely worthwhile trying to start with a fresh cache entry. You could do this by running the build with `cache-write-only: true`, since that should start with an empty Gradle User Home.

> I've enabled that variable and can see that it is true in the debug logs, but the makeup of the gradle user home was not output.

To clarify, you won't see this for the failing case because the Gradle User Home entry is not downloaded successfully.
But if you start with a fresh cache entry, you can use the cache entry logging to get a feel for what's taking up so much room in the cache entry. I suspect that the Gradle User Home is growing over time until it reaches a  size that cannot be handled but GitHub actions.

Are you able to share links to the workflow run? I'd be happy to look deeper if I can see the outputs.

**Comment by bigdaz** [comment]
The underlying error appears to be due to https://github.com/actions/cache/issues/773. Once a fixed version of `actions/cache` is released I'll ask you to re-test this.

**Comment by bigdaz** [comment]
This should be resolved with `v2.1.5`, due to upgrade to `@actions/cache` `v2.0.0`.
